layout/_macro/post.swig

```diff
@@ -59,7 +59,12 @@
                 <a class="post-title-link" href="{{ url_for(post.path) }}" itemprop="url">{#
                 #}{{ post.title | default(__('post.untitled'))}}{#
               #}</a>{#
-            #}{% else %}{{ post.title }}{% endif %}{#
+            #}{% else %}
+              {{ post.title }}
+              {% if theme.post_edit.enable %} 
+                <a href="{{theme.post_edit.url}}{{post.source.slice(7)}}" class="article-edit-link" title="{{ __('post.edit') }}"><i class="fa fa-pencil"></i></a>    
```

Comment:
Can i ask what's mean `post.source.slice(7)`? I'm interesting in `slice(7)` parameter.

Reply:
in the above example
```
post.source = `_post/web/blog-framework/nodejs-hexo/plugin/hexo-plugin-recommendation.md`
```

`post.source.slice(7) = web/blog-framework/nodejs-hexo/plugin/hexo-plugin-recommendation.md`

Reply:
I am not sure whether all the `post.source` starts with `_post/`. Maybe need to be optimized.